test/libyul/EVMDialectCompatibility.h

```diff
@@ -0,0 +1,78 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <libyul/backends/evm/EVMDialect.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <fmt/format.h>
+
+#include <cstdint>
+#include <optional>
+#include <vector>
+
+
+namespace solidity::yul::test
+{
+
+inline void requireBuiltinCompatibility(EVMDialect const& _dialect1, EVMDialect const& _dialect2, std::string_view const _builtin)
+{
+	if (auto const currentHandle = _dialect1.findBuiltin(_builtin))
+		if (auto const handle = _dialect2.findBuiltin(_builtin))
```

Comment:
This will ensure that builtins available in both dialects have matching IDs, which is necessary but not sufficient for full compatibility. We also need to ensure that the dialect with objects has all the builtins present in the one without.